lib/spack/spack/__init__.py

```diff
@@ -188,7 +188,7 @@
 __all__ += spack.util.executable.__all__
 
 # User's editor from the environment
-editor = Executable(os.environ.get("EDITOR", "vi"))
+editor = Executable(os.environ.get("EDITOR", "vim"))
```

Comment:
I don't know.  But maybe some embedded or stripped down ones (e.g. for containers)?  What about:

```python
if "EDITOR" in os.environ:
    editor = Executable(os.environ.get("EDITOR"))
else:
    editor = which('vim')
    if not editor:
        editor = Executable('vi')
```

I'm an `emacs` user -- I just picked `vi` as the default because it's more likely to be installed, so I guess we should do best effort here.